mlflow/pyfunc/__init__.py

```diff
@@ -224,6 +224,8 @@ def predict(*args):
         columns = [str(i) for i, _ in enumerate(args)]
         pdf = pandas.DataFrame(schema, columns=columns)
         result = model.predict(pdf)
+        if type(result) == pandas.DataFrame:
+            return result[result.columns[0]]
```

Comment:
Why do we select the the first column from the output dataframe; do we always expect that this dataframe will only have a single column? If so, we should validate that this is the case and print a useful error message if the dataframe does not conform to our specifications. A comment explaining this restriction would also be useful.

Reply:
Currently spark_udf can only handle 1-d output. We plan to address that in the future but for now I think returning partial result is better than failing.